src/ui/public/notify/banners/banners.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Banners represents a prioritized list of displayed components.
+ */
+export class Banners {
+
+  constructor() {
+    // sorted in descending order (100, 99, 98...) so that higher priorities are in front
+    this.list = [];
+    this.onChangeCallback = null;
+  }
+
+  _changed = () => {
+    if (this.onChangeCallback) {
+      this.onChangeCallback();
+    }
+  }
+
+  _replaceIfExists = bannerDetails => {
```

Comment:
This becomes moot if we implement my suggestion of assigning an ID internally.

Reply:
Since it's still allowed, it remained.